solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -368,6 +370,86 @@ contract StandardPoolConverter is
         return reserveBalances(1, 2);
     }
 
+    /**
+     * @dev transfers a portion of the accumulated conversion fees
+     */
+    function transferFees() external protected {
+        (uint256 reserveBalance0, uint256 reserveBalance1) = transferFeesAndSyncReserveBalances(0);
+        __reserveBalancesProd = reserveBalance0 * reserveBalance1;
+    }
+
+    /**
+     * @dev transfers a portion of the accumulated conversion fees and syncs the reserve balances
+     *
+     * @param _value    amount of ether to exclude from the ether reserve balance (if relevant)
+     * @return new reserve balances
+     */
+    function transferFeesAndSyncReserveBalances(uint256 _value) internal returns (uint256, uint256) {
+        syncReserveBalances(_value);
+
+        (uint256 reserveBalance0, uint256 reserveBalance1) = reserveBalances(1, 2);
+        (uint256 currPoint, uint256 prevPoint) = networkFeePoints(reserveBalance0 * reserveBalance1, __reserveBalancesProd);
+
+        if (currPoint > prevPoint) {
+            INetworkSettings networkSettings = INetworkSettings(addressOf(NETWORK_SETTINGS));
+            (address networkFeeWallet, uint32 networkFee) = networkSettings.networkFeeParams();
+            (uint256 n, uint256 d) = networkFeeRatio(currPoint, prevPoint, networkFee);
+            uint256 fee0 = reserveBalance0.mul(n).div(d);
+            uint256 fee1 = reserveBalance1.mul(n).div(d);
+            reserveBalance0 -= fee0;
+            reserveBalance1 -= fee1;
+            setReserveBalances(1, 2, reserveBalance0, reserveBalance1);
+            safeTransfer(__reserveTokens[0], networkFeeWallet, fee0);
+            safeTransfer(__reserveTokens[1], networkFeeWallet, fee1);
+        }
+
+        return (reserveBalance0, reserveBalance1);
+    }
+
+    /**
+     * @dev returns the reserve balances of the given reserve tokens minus their corresponding fees
+     *
+     * @param _reserveTokens    reserve tokens
+     * @return reserve balances minus their corresponding fees
+     */
+    function reserveBalancesMinusFees(IERC20[] memory _reserveTokens) internal view returns (uint256[2] memory) {
```

Comment:
Fixed.

Reply:
Unfixed; This requires changing many other functions in the file. I will open a separate task for it.

Reply:
@lbeder : from [the official docimentation](https://docs.soliditylang.org/en/v0.5.0/types.html?highlight=calldata#data-location):
> Calldata is only valid for parameters of external contract functions and is required for this type of parameter. Calldata is a non-modifiable, non-persistent area where function arguments are stored, and behaves mostly like memory.

The function above is internal, so I don't think that `calldata` is even permitted in this case.

Reply:
You can use `calldata` for `internal` or `private` function as well. It will work well for functions that receive a propagated "read-only" arguments from `external` or `public` function, so should be applicable in some of these cases. In any case, keep this in mind during refactoring of existing/new code.